tests/runtime/bool

```diff
@@ -0,0 +1,49 @@
+NAME bool values
+PROG BEGIN { print((true, false)); exit(); }
+EXPECT (true, false)
+TIMEOUT 1
+
+NAME bool in conditional
+PROG BEGIN { if (true) { print(1); } if (false) { print(2); } exit(); }
+EXPECT 1
+EXPECT_NONE 2
+TIMEOUT 1
+
+NAME bool as map keys and values
+PROG BEGIN { @a[true] = false; exit(); }
+EXPECT @a[true]: false
+TIMEOUT 1
+
+NAME bool as a variable
+PROG BEGIN { $b = true; $c = false; print(($b, $c)); exit(); }
+EXPECT (true, false)
+
+NAME cast int to bool
+PROG BEGIN{ print(((bool)1, (bool)0)); exit(); }
+EXPECT (true, false)
+TIMEOUT 1
+
+NAME cast string to bool
+PROG BEGIN{ print(((bool)"hello", (bool)"")); exit(); }
+EXPECT (true, false)
+TIMEOUT 1
+
+NAME cast ptr to bool
+PROG BEGIN{ $a = (int64*)0; print(((bool)percpu_kaddr("process_counts", 0), (bool)$a)); exit(); }
```

Comment:
nit: is the `percpu_kaddr` necessary here? It'd just be nice to not have a dependency on the kernel symbol `process_counts` if we don't need it. Would something like `$a = (int64*)0; $b = (int64*)1; print((bool)$a, (bool)$b))` be enough?

Also it looks like `percpu_kaddr` returns a `uint64`, not a pointer type after all